agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -234,245 +134,318 @@ public CallReplacementTransformer() {}
+                invocation.getArgumentTypes(pgen),
+                Const.INVOKESTATIC));
+        break;
 
       default:
-        return (null);
+        return null;
+    }
+    return instructionList;
+  }
+
+  /**
+   * Returns the replacement method for the given method if one is determined by a method, class or
+   * package replacement, or null otherwise.
+   *
+   * <p>Class or package replacements are represented as strings in the {@link
+   * #prefixReplacementMap}. When the argument method belongs to one of these classes/packages, a
+   * new {@link MethodDef} replacement is constructed for the method and is added to the {@link
+   * #replacementMap}.
+   *
+   * @param orig the method to replace, must not be null
+   * @return the replacement method, null if there is none
+   */
+  private MethodDef getReplacement(MethodDef orig) {
+    MethodDef replacement = replacementMap.get(orig);
+    if (replacement != null) {
+      return replacement;
+    }
+    // check for a class or package prefix.
+    String prefix = orig.getClassname();
+    String prefixReplacement = prefixReplacementMap.get(prefix);
+    // if prefixReplacement not null, the class name has a replacement
+    // otherwise, strip off classname and search for package
+    if (prefixReplacement == null) {
+      int dotPos = prefix.lastIndexOf('.');
+      if (dotPos > 0) {
+        prefix = prefix.substring(0, dotPos);
+        prefixReplacement = prefixReplacementMap.get(prefix);
+      }
+    }
+    // if prefixReplacement is not null, then use it to replace prefix
+    // in class name to create replacement method name
+    if (prefixReplacement != null) {
+      String replacementName = prefixReplacement + orig.getClassname().substring(prefix.length());
+      replacement = new MethodDef(replacementName, orig.getName(), orig.getArgTypes());
+      replacementMap.put(orig, replacement);
     }
+
+    return replacement;
   }
 
   /**
-   * Replace instruction ih in list il with the instructions in new_il. If new_il is null, do
-   * nothing.
+   * Replace the instruction in the original instruction list with the instructions in the new
+   * instruction list.
    *
-   * @param il the instruction list
-   * @param ih the instruction
-   * @param new_il the new instructions to substitute
+   * @param instructionList the original instruction list
+   * @param instruction the instruction to replace
+   * @param newInstructionList the new instructions to substitute for the instruction, must be
+   *     non-null and non-empty
    */
-  protected static void replace_instructions(
-      InstructionList il, InstructionHandle ih, InstructionList new_il) {
+  private static void replaceInstructions(
+      InstructionList instructionList,
+      InstructionHandle instruction,
+      InstructionList newInstructionList) {
 
-    if ((new_il == null) || new_il.isEmpty()) {
+    if ((newInstructionList == null) || newInstructionList.isEmpty()) {
       return;
```

Comment:
The documentation states that `newInstructionList` must be non-null and non-empty.  When the precondition is violated, it's better style and less error-prone to throw an exception rather than silently returning.